### ISSUE TITLE
crash on long string [YAML #90593] [rt.cpan.org #92544]

### ISSUE THREAD

**Issue by ingydotnet** [ingydotnet/yaml-pm#134]
https://rt.cpan.org/Ticket/Display.html?id=92544

```
RT#90593 claims that this bug was fixed in version 0.85.  But I am still 
encountering this problem in YAML version 0.88, using Perl 5.8.8.

Simple test: just run t/rt-90593.t.  It crashes at line 518, during the 
invocation of $self->inline.  This indicates a problem with the inline() 
function.  The crash does not occur at that location when I replace 
$self->inline with $self->{inline}.

Thanks and best wishes,
--Jonathan
```

**Comment by eserte** [comment]
I just got the following failure with perl 5.25.4 on a freebsd 10.1 system:
```
t/rt-90593.t .............. 
Failed 2/2 subtests 
...
t/rt-90593.t            (Wstat: 9 Tests: 0 Failed: 0)
  Non-zero wait status: 9
```
9 means SIGKILL, so it's likely that this was also caused by a segmentation fault or something similar. It seems that the problem is rare; right now I cannot reproduce it.

**Comment by abrahamdsl** [comment]
I also had this problem. In case it might be useful to someone, I closed all other programs running in my system to free up memory and the test was successful. Basing on experiences with SIGKILL, I had hypothesised that the error is due to lack of memory which was again correct in my case.

**Comment by perlpunk** [comment]
Released YAML 1.24_001
